examples/basic_authentication/lib/user_repository.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+final List<User> _db = [];
```

Comment:
```suggestion
final List<User> _database = [];
```

I would avoid the abbreviation to avoid confusion. Something similar is also mentioned on [Effective Dart](https://dart.dev/effective-dart/documentation#avoid-abbreviations-and-acronyms-unless-they-are-obvious).

Reply:
I also try to avoid abbreviations everytime that is possible, but in this case, I will argue that this is still compliant to effective dart, as we see there:

> AVOID abbreviations and acronyms unless they are obvious

And `db` is super common and pretty obvious term for database abbreviations.

Reply:
My brain goes through an additional process when reading "db". I read "db" and then translate it to "database" and then continue reading. Whereas, my brain does not do the same when reading "CLI".

I agree "db" is common and hence this would also fit within the Effective Dart guideline. So the change is just personal preference. Feel free to resolve this Nit.